DERIVED_DM_KEYS.md

```diff
@@ -0,0 +1,38 @@
+# Derived DM Keys
+
+The ssb-meta-feeds-spec defines how and where to announce the public part of your DM keys
+for particular leaf feeds, but this leaves you with the challenge of how and where to persist
+the private parts of those keys.
+
+This sub-spec defines a way to deterministically derive your DM keys from your meta-feed seed.
+
+1. when annoucning a leaf feed, we generate a `nonce` (32 bytes of random data)
+2. we combine this with our `seed` to generate a curve25519 keypair
+3. we publish the public key AND the `nonce` in the announce message metadata
+  ```js
+  {
+    type: 'meta/add/derived',
+    purpose: 'chess',
+    subfeed: 'ssb:feed/classic/DIoOBMaI1f0mJg+5tUzZ7vgzCeeHh8+zGta4pOjc+k0='
+    metadata: {
+      nonce: '9OaY/mnPaWnu+kOfxBtFRgQLgXyq3X2yEOW48jDrw='
+      encryption: {
+        curve: 'curve25519',
+        public: 'BtFRgQLgXyq3G48jDrX9OaY/mnPaWnu+kOfx2yEOWwU='
+      }
+    }
+  }
+  ```
+
+## 1. Deriving an encryption key
+
+```js
+const nonce = crypto.randomBytes(32)
+
+function deriveDiffieHellmanKeys (seed, nonce) {
+
+  return hkdf.expand(
```

Comment:
Suggestion, like this: https://github.com/ssbc/ssb-meta-feeds/blob/58454e8440eab0bd64ed0bf63c6cd9a9c8aaf5d1/keys.js#L51-L55

Using the prefix `'ssb-meta-feeds-dm-v1:'` and the salt `ssb`.

Reply:
yes something like that seems reasonable

This PR is on hold till other things are done I think

Reply:
> This PR is on hold till other things are done I think

Which other things? This spec PR is blocking me from implementing the announcements in ssb-box2, ssb-meta-feeds, ssb-tribes2

Reply:
done